_posts/2021-04-12-homebrew-3.1.0.md

```diff
@@ -0,0 +1,45 @@
+- [Various deprecated, disabled and deleted code.](https://github.com/Homebrew/brew/pull/11075)
+- [The undocumented `HOMEBREW_NO_ENV_FILTERING` flag is deprecated and will be removed.](https://github.com/Homebrew/brew/pull/10778)
+
+- [The `HOMEBREW_NO_BOTTLE_SOURCE_FALLBACK` is removed (as its behaviour is now the default).](https://github.com/Homebrew/brew/pull/10640)
+- [GitHub Packages writes bottles' tabs into the GitHub Packages manifest JSON. `brew bottle --only-json-tab` has been added to allow future GitHub Packages bottles to store their tab outside the bottle to allow reproducible builds.](https://github.com/Homebrew/brew/pull/10948) [This is why downloads from GitHub Packages also include a small manifest JSON download.](https://github.com/Homebrew/brew/pull/10977) [This also allows future creation of `all: SHA256` bottles for bottles shared across all platforms.](https://github.com/Homebrew/brew/pull/11077)
+
+Other changes since 3.0.0 I'd like to highlight are the following:
+
+- [`HOMEBREW_BOOTSNAP` is enabled by default for Homebrew developers.](https://github.com/Homebrew/brew/pull/10680)
+- [The `sed` superenv shim is removed.](https://github.com/Homebrew/brew/pull/10802)
+- [CodeCov in Homebrew's GitHub Actions no longer requires a hardcoded token.](https://github.com/Homebrew/brew/pull/10514)
+- [`brew update --quiet` is a bit quieter.](https://github.com/Homebrew/brew/pull/10868)
+- [`brew update` outputs to `stderr` when not outputting to a TTY to ease `stdout` usage.](https://github.com/Homebrew/brew/pull/10501)
+- [`brew update` reports outdated formulae and suggests `brew upgrade`.](https://github.com/Homebrew/brew/pull/10581)
+- [`installed_on_request` is correctly set in the tab for dependencies again. This will improve usage of `brew bundle dump` after reinstalling packages.](https://github.com/Homebrew/brew/pull/10681)
+- [Homebrew does not load non-Bundler gems to prevent issues from user-installer gems.](https://github.com/Homebrew/brew/pull/10695)
+- [Homebrew's default `ruby` warning level is now `-W1` (to avoid hiding legitimate warnings.)](https://github.com/Homebrew/brew/pull/10598)
+- [`brew cask` `pkg` uninstallation is faster.](https://github.com/Homebrew/brew/pull/10954)
+- [`brew --prefix <formula>` is faster.](https://github.com/Homebrew/brew/pull/10693)
+- [`brew --version` is faster.](https://github.com/Homebrew/brew/pull/10771)
+- [`brew list` uses `ls` more for formulae, casks and more consistent output.](https://github.com/Homebrew/brew/pull/10945)
+- [`brew list` visually separates formulae and casks.](https://github.com/Homebrew/brew/pull/10899)
+- [`brew bump` supports casks.](https://github.com/Homebrew/brew/pull/10635)
+- [`service do` is a new formula DSL to allow plists to be easily generated.](https://github.com/Homebrew/brew/pull/9517)
+- [Always require deveveloper tools on Apple Silicon (because `codesign` is needed when pouring bottles).](https://github.com/Homebrew/brew/pull/10533)
```

Comment:
```suggestion
- [Always require developer tools on Apple Silicon (because `codesign` is needed when pouring bottles).](https://github.com/Homebrew/brew/pull/10533)
```

Reply:
deveveloper is the new covfefe